### PR TITLE
Fix #1605: don't inline methods that have errors

### LINKED ISSUE #1605: Inlining invalid method crashes compiler
With:

``` scala
object Test {
  def foo = inlineMe

  inline def inlineMe = 1 + x // where x is not defined anywhere
}
```

the compiler crashes.

### PR DESCRIPTION
**Pull request by felixmulder**
Review: @smarter

### CODE REVIEW

**Comment by DarkDimius** [comment]
nice title)

**Comment by felixmulder** [comment]
It inlined it ;)

**Comment by smarter** [comment]
I think I would put the check at the place where we call `makeInlineable` in `registerInlineInfo` instead of deep into `makeInlineable` itself.

**Comment by felixmulder** [comment]
@smarter - done

File: src/dotty/tools/dotc/typer/Inliner.scala [hunk 1]
```diff
@@ -191,7 +191,9 @@ object Inliner {
           sym.updateAnnotation(LazyBodyAnnotation { _ =>
             implicit val ctx: Context = inlineCtx
             val tree1 = treeExpr(ctx)
```
Comment (odersky) [inline src/dotty/tools/dotc/typer/Inliner.scala]: Better use reporter.errorCount

File: src/dotty/tools/dotc/typer/Inliner.scala [hunk 2]
```diff
@@ -190,8 +190,7 @@ object Inliner {
           val inlineCtx = ctx
           sym.updateAnnotation(LazyBodyAnnotation { _ =>
             implicit val ctx: Context = inlineCtx
-            val tree1 = treeExpr(ctx)
-            makeInlineable(tree1)
+            ctx.withNoError(treeExpr(ctx))(makeInlineable)
```
Comment (felixmulder) [inline src/dotty/tools/dotc/typer/Inliner.scala]: @odersky - updated with your suggestion

**Comment by odersky** [comment]
LGTM

**Pull request merged**